src/Microsoft.AspNetCore.Mvc.Razor.Host/ModelExpressionPass.cs

```diff
@@ -22,7 +24,9 @@ private class Visitor : RazorIRNodeWalker
 
             public override void VisitSetTagHelperProperty(SetTagHelperPropertyIRNode node)
             {
-                if (string.Equals(node.Descriptor.TypeName, "Microsoft.AspNetCore.Mvc.ViewFeatures.ModelExpression", StringComparison.Ordinal))
+                if (string.Equals(node.Descriptor.TypeName, ModelExpressionTypeName, StringComparison.Ordinal) ||
+                    (node.IsIndexerNameMatch &&
+                     string.Equals(node.Descriptor.IndexerTypeName, ModelExpressionTypeName, StringComparison.Ordinal)))
```

Comment:
Was this a bug? Should we have a test for this case? This would be like `IDictionary<string, ModelExpression>`